linera-views/src/reentrant_collection_view.rs

```diff
@@ -584,17 +605,33 @@ where
                 btree_map::Entry::Occupied(entry) => {
                     let entry = entry.into_mut();
                     if let Update::Removed = entry {
-                        selected_short_keys.push((short_key, true));
+                        let key = self
+                            .context
+                            .base_tag_index(KeyTag::Subview as u8, &short_key);
+                        let context = self.context.clone_with_base_key(key);
+                        let view = W::new(context)?;
+                        let view = Arc::new(RwLock::new(view));
```

Comment:
This code is duplicated a lot. Maybe create a method for the first two lines at least?

Reply:
I could refactor the code into something much nicer and eliminate the code duplication.